plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+    if sandbox.get('services_created'):
+      actions_dict['Services created'] = sandbox.get('services_created')
+    if sandbox.get('services_started'):
+      actions_dict['Services started'] = sandbox.get('services_started')
+    if sandbox.get('services_stopped'):
+      actions_dict['Services stopped'] = sandbox.get('services_stopped')
+    if sandbox.get('services_deleted'):
+      actions_dict['Services deleted'] = sandbox.get('services_deleted')
+
+    if sandbox.get('windows_searched'):
+      actions_dict['Windows searched'] = sandbox.get('windows_searched')
+    if sandbox.get('windows_hidden'):
+      actions_dict['Windows hidden'] = sandbox.get('windows_hidden')
+
+    if sandbox.get('mutexes_opened'):
+      actions_dict['Mutexes opened'] = sandbox.get('mutexes_opened')
+    if sandbox.get('mutexes_created'):
+      actions_dict['Mutexes created'] = sandbox.get('mutexes_created')
+
+    if sandbox.get('crypto_algorithms_observed'):
+      actions_dict['Crypto alg. observed'] = sandbox.get('crypto_algorithms_observed')
+    if sandbox.get('crypto_keys'):
+      actions_dict['Crypto keys'] = sandbox.get('crypto_keys')
+    if sandbox.get('crypto_plain_text'):
+      actions_dict['Crypto plain texct'] = sandbox.get('crypto_plain_text')
+
+    if sandbox.get('text_decoded'):
+      actions_dict['Text decoded'] = sandbox.get('text_decoded')
+    if sandbox.get('text_highlighted'):
+      actions_dict['Text highlighted'] = sandbox.get('text_highlighted')
+
+    if sandbox.get('ja3_digests'):
+      actions_dict['JA3 digests'] = sandbox.get('ja3_digests')
+
+    if sandbox.get('modules_loaded'):
+      actions_dict['Modules loaded'] = sandbox.get('modules_loaded')
+
+    if sandbox.get('registry_keys_opened'):
+      actions_dict['Reg. keys opened'] = sandbox.get('registry_keys_opened')
+    if sandbox.get('registry_keys_set'):
+      # Returns a list of dictionaries
+      actions_dict['Reg. keys set'] = sandbox.get('registry_keys_set')
+    if sandbox.get('registry_keys_deleted'):
+      actions_dict['Reg. keys deleted'] = sandbox.get('registry_keys_deleted')
+
+    if sandbox.get('ids_alerts'):
+      # Returns a list of dictionaries
+      actions_dict['IDS alerts'] = sandbox.get('ids_alerts')
+
+    return actions_dict
+
+  def get_comments(self):
+    return self.comments
+
+  def get_contacted_ips(self):
+    return self.contacted_ips.keys()
+
+  def get_contacted_ip(self, ip):
+    """Returns a list containing ip + av detections ratio."""
+
+    ip_report = self.contacted_ips[ip]
+    ip_details = []
+    ip_analysis = ip_report.get('last_analysis_stats')
+    total = ip_analysis['harmless'] + ip_analysis['malicious']
+    total += ip_analysis['suspicious'] + ip_analysis['undetected']
+    ip_analysis = str(ip_analysis['malicious']) + '/' + str(total)
+    ip_details.append(ip)
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
```

Comment:
In functions like these, I'd typically name a variable "result" and that's the one which would be returned. This makes very easy to eyeball the code and find out if you are mistakenly reading from the result instead of reading from the original data. One again, this is mostly a "prevent bugs from appearing later on" suggestion.